src/main/java/com/commercetools/sync/services/impl/TypeServiceImpl.java

```diff
@@ -33,13 +53,62 @@ public TypeServiceImpl(@Nonnull final BaseSyncOptions syncOptions) {
     @Override
     public CompletionStage<Optional<String>> fetchCachedTypeId(@Nonnull final String key) {
         if (!isCached) {
-            return fetchAndCache(key);
+            return cacheAndFetch(key);
```

Comment:
Wasn't the name more correct? With the correct order of events fetch -> cache?

Reply:
You are right :  https://github.com/commercetools/commercetools-sync-java/pull/333/commits/76c5a44038422f2c25179c48cd5e082f45b66c18